### ISSUE TITLE
Timeout implementation doesn't match IO::KQueue

### ISSUE THREAD

**Issue by TLINDEN** [karupanerura/p5-Filesys-Notify-KQueue#1]
Hi,

(this is the same as reported in https://rt.cpan.org/Ticket/Display.html?id=104247)

the IO::KQueue pod says about the kevent() timeout parameter:

> Timeout is in milliseconds. If timeout is ommitted then we wait
> forever until there are events to read. If timeout is zero then
> we return immediately.

So I used a timeout of 0 like:

```
  my $notify = Filesys::Notify::KQueue->new(
                        path    => [$dir],
                        timeout => 0,
                       );
```

However, the wait() call never returned. The problem was, that the parameter was ignored if 0:

```
  $self->timeout($args->{timeout} || $class->default_timeout);
```

But after changing this, it still didn't work because of this:

```
  sub wait {
    my ($self, $cb) = @_;

    my $events = $self->get_events;
    until (@$events) {
        $events = $self->get_events;
    }
  [..]
```

It runs endless til some event occurs, which is not what I wanted.

So, the attached patch fixes both issues and allows to use a timeout of 0, which causes wait() to return immediately, with or without events.

However, there's still an issue with the timeout. Because, according to the mentioned kevent() documentation I'd expect wait() to return after the given timeout has been reached, or to return immediately if set to 0, or to never return if no timeout was given.

The current implementation does never return, with the supplied patch it at least implements timeout zero, but this needs to be reworked, I think, as a whole. In short, this is what I would expect:

```
 timeout => 0     # return immediately
 timeout => >1    # return after timeout
 timeout => undef # never return
```

[the patch is attached to the RT bug]

best regards,
Tom

**Comment by karupanerura** [comment]
Thank you for reporting!
I'll read it in this weekend. (and fix it, maybe.)

**Comment by karupanerura** [comment]
I released it as version 0.10. Many thanks!
